### PR TITLE
Push workdir state on file save

### PR DESCRIPTION
**Pull request by krlvi**
related https://github.com/sturdy-dev/sturdy/pull/11

### CODE REVIEW

**Comment by krlvi** [comment]
probably better not to initialize everything inside work, hopefully more stuff can be event driven

**Comment by krlvi** [comment]
@zegl ptal

File: src/api.ts [hunk 1]
```diff
@@ -16,3 +18,17 @@ export function headersWithAuth(token: string): Headers {
         "x-client-version": version,
     };
 }
+
+export const postWorkDirForRepo = async (conf: Configuration | undefined, owner: string, name: string, workingTreeDiff: string, head: string) => {
```
Comment (zegl) [inline src/api.ts:22]: Why does this function take `Configuration | undefined`, when it's immediately checked if `conf` is truthy?

Comment (zegl) [reply src/api.ts:22]: Should this function be in some other file maybe? It's not related to the general API functions in this file.

File: src/extension.ts [hunk 2]
```diff
@@ -22,6 +25,38 @@ export function activate(context: vscode.ExtensionContext) {
...
   })
+
+  // Push work dir
+  vscode.workspace.onDidSaveTextDocument(async () => {
+      let git = initGit();
+      if (!git) return
+      let conf: Configuration | undefined = vscode.workspace.getConfiguration().get("conf.sturdy");
+      if (!conf) return
+      let repos = await LookupConnectedSturdyRepositories(git, conf);
+      if (!repos) return
```
Comment (zegl) [inline src/extension.ts:36]: Performing these actions on _every_ save seems wasteful. Can these values be initialised beforehand?

File: src/extension.ts [hunk 3]
```diff
@@ -22,6 +25,38 @@ export function activate(context: vscode.ExtensionContext) {
...
+      let workingTreeDiff = await git.diff()
+      let head = await git.revparse("HEAD");
+      repos.repos.forEach((r) => api.postWorkDirForRepo(conf, r.owner, r.name, workingTreeDiff, head))
+  })
+}
+
+function initGit(): SimpleGit | undefined {
```
Comment (zegl) [inline src/extension.ts:43]: There already is a function called `initGit`, could we use that one instead?

**Comment by krlvi** [comment]
https://github.com/sturdy-dev/sturdy-vscode/pull/13

**Pull request closed without merging**